src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementBackedDocumentsService.java

```diff
@@ -98,6 +97,28 @@ private Claim getClaimByExternalId(String externalId, String authorisation) {
         return claimService.getClaimByExternalId(externalId, authorisation);
     }
 
+    private byte[] processRequest(Claim claim,
+                                  String authorisation,
+                                  ClaimDocumentType claimDocumentType,
+                                  PdfService pdfService,
+                                  String baseFileName) {
+        Optional<URI> claimDocument = claim.getClaimDocument(claimDocumentType);
+        try {
+            if (claimDocument.isPresent()) {
+                URI documentSelfPath = claimDocument.get();
+                return documentManagementService.downloadDocument(authorisation, documentSelfPath, baseFileName);
+            } else {
+                PDF document = new PDF(baseFileName, pdfService.createPdf(claim), claimDocumentType);
+                uploadToDocumentManagement(document,
+                    authorisation,
+                    claim);
+                return document.getBytes();
+            }
+        } catch (Exception ex) {
+            return pdfService.createPdf(claim);
```

Comment:
can we isolate the pdfgeneration exception and handle it differently? Because if the pdfService.createPdf failed, then handling it with same call wont do any better. Just suggestion

Reply:
basically what ever the exception.. we want to return the document to the user..

Reply:
but if we have exception from pdfservice also we attempt to call the same. Can we have two catch statements is my question.